warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py

```diff
@@ -423,36 +448,59 @@ def _get_browser_creation_function(self, browser_name):
         func_name = BROWSER_NAMES.get(browser_name.lower().replace(' ', ''))
         return getattr(self, func_name) if func_name else None
 
-
-    def _make_ff(self, webdriver_remote_url, desired_capabilites, profile_dir):
+    def _make_ff(self, webdriver_remote_url, desired_capabilites, profile_dir,
+                 binary, gecko_path):
         """Create an instance of firefox browser"""
-
-        if webdriver_remote_url:
-            browser = self._create_remote_web_driver(webdriver.DesiredCapabilities.FIREFOX,
-                                                     webdriver_remote_url, desired_capabilites,
-                                                     profile_dir)
-        else:
-            browser = webdriver.Firefox(firefox_profile=profile_dir)
-        return browser
-
-    def _make_chrome(self, webdriver_remote_url, desired_capabilities, profile_dir):
+        try:
+            if webdriver_remote_url:
+                browser = self._create_remote_web_driver(
+                        webdriver.DesiredCapabilities.FIREFOX,
+                        webdriver_remote_url, desired_capabilites,
+                        profile_dir)
+            else:
+                ff_capabilities = webdriver.DesiredCapabilities.FIREFOX
+                if ff_capabilities['marionette']:
+                    ff_capabilities['acceptInsecureCerts'] = True
+                    ffbinary = FirefoxBinary(binary)
+                    browser = webdriver.Firefox(firefox_binary=ffbinary,
+                                                firefox_profile=profile_dir,
+                                                executable_path=gecko_path)
+                else:
+                    browser = webdriver.Firefox(firefox_profile=profile_dir)
+            return browser
+        except WebDriverException as e:
+            if "executable needs to be in PATH" in str(e):
+                print_error("Please provide path for geckodriver executable")
+            elif "Expected browser binary location" in str(e):
+                print_error("Please provide path of firefox executable")
+
+    def _make_chrome(self, webdriver_remote_url, desired_capabilities,
+                     profile_dir, binary, gecko_path):
```

Comment:
No explanation about gecko_path appearance in here
chrome doesn't need gecko_path and this is passed with no use